tests/test_application.py

```diff
@@ -1287,6 +1287,24 @@ async def callback_2():
             await asyncio.sleep(0.05)
             assert stop_task.done()
 
+    async def test_create_task_awaiting_future(self, app):
+        async def callback():
+            await asyncio.sleep(0.01)
+            return 42
+
+        out = await app.create_task(asyncio.gather(callback()))
+        assert out == [42]
+
+    async def test_create_task_awaiting_generator(self, app):
+        event = asyncio.Event()
+
+        def gen():
+            yield
+            event.set()
+
+        await app.create_task(gen())
```

Comment:
can we also check a return value here?

Reply:
We can do that if we use `asyncio.coroutine` decorator but it prints out deprecated warnings like hell for python>=3.8
Generator-based coroutine functions in asyncio has been kept  only for backward compatibility and you should drop it after 4 months (Python 3.7 EOL)

Reply:
All right, let's skip it, then